src/Compilers/CSharp/Portable/Binder/ForLoopBinder.cs

```diff
@@ -72,8 +72,14 @@ private BoundForStatement BindForParts(ForStatementSyntax node, Binder originalB
             // Declaration and Initializers are mutually exclusive.
             if (_syntax.Declaration != null)
             {
-                ImmutableArray<BoundLocalDeclaration> unused;
-                initializer = originalBinder.BindForOrUsingOrFixedDeclarations(node.Declaration, LocalDeclarationKind.RegularVariable, diagnostics, out unused);
+                var type = _syntax.Declaration.Type is ScopedTypeSyntax scopedType
+                    ? scopedType.Type
+                    : _syntax.Declaration.Type;
```

Comment:
`_syntax.Declaration.Type.SkipScoped(out _)`

Reply:
Done.  Dumb question, is there no feature flag check needed here (like we have for the ref-type below?)

Reply:
> is there no feature flag check needed here (like we have for the ref-type below?)

It looks like the feature check is in `BindForOrUsingOrFixedDeclarations()`.

Reply:
thanks!